pkg/i2gw/providers/gce/gce_extensions.go

```diff
@@ -0,0 +1,201 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gce
+
+import (
+	"context"
+	"encoding/json"
+
+	gkegatewayv1 "github.com/GoogleCloudPlatform/gke-gateway-api/apis/networking/v1"
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw"
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw/ir"
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw/notifications"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	backendconfigv1 "k8s.io/ingress-gce/pkg/apis/backendconfig/v1"
+	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"
+	"sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+type serviceNames []types.NamespacedName
+
+func getBackendConfigMapping(ctx context.Context, storage *storage) map[types.NamespacedName]serviceNames {
+	beConfigToSvcs := make(map[types.NamespacedName]serviceNames)
+
+	for _, service := range storage.Services {
+		svc := types.NamespacedName{Namespace: service.Namespace, Name: service.Name}
+		ctx = context.WithValue(ctx, serviceKey, service)
+
+		// Read BackendConfig based on v1 BackendConfigKey.
+		beConfigName, exists := getBackendConfigName(ctx, service, backendConfigKey)
+		if exists {
+			beConfigKey := types.NamespacedName{Namespace: service.Namespace, Name: beConfigName}
+			beConfigToSvcs[beConfigKey] = append(beConfigToSvcs[beConfigKey], svc)
+			continue
+		}
+
+		// Read BackendConfig based on v1beta1 BackendConfigKey.
```

Comment:
Pretty sure you've already covered this, but just making sure - does ingress-gce also give precedence to the v1 key over the v1beta1 one?

Reply:
Sorry I think I missed the previous comment. v1 key has precedence over v1beta1(See [here](https://github.com/kubernetes/ingress-gce/blob/2c4530cdd9e9597e0044800c2c95039cbc0f2e73/pkg/annotations/service.go#L443)), and we are doing the same thing here: If we found with v1 key on this service, we stop looking for v1beta1 key and move on to the next one, which is consistent.